Modules/Devices/src/Devices.Application/Clients/Commands/CreateClients/Handler.cs

```diff
@@ -22,12 +27,27 @@ public async Task<CreateClientResponse> Handle(CreateClientCommand request, Canc
                 throw new ApplicationException(ApplicationErrors.Devices.ClientIdAlreadyExists());
         }
 
+        if (!string.IsNullOrEmpty(request.TierId))
+        {
+            var tierIdResult = TierId.Create(request.TierId);
+            if (tierIdResult.IsFailure)
+                throw new ApplicationException(ApplicationErrors.Devices.InvalidTierId());
+
+            _ = await _tiersRepository.FindById(tierIdResult.Value, cancellationToken) ?? throw new ApplicationException(GenericApplicationErrors.NotFound(nameof(Tier)));
```

Comment:
This is so ugly...
1. You should always request as little data as possible from the database. In this case, all you need is true or false (does the Tier exist or not)
2. Assigning the result of the Find method to a variable just so you can use the `??` operator is a hack and should be avoided.